diff --git a/lib/internal/Magento/Framework/Filesystem/Directory/PathValidator.php b/lib/internal/Magento/Framework/Filesystem/Directory/PathValidator.php
index 5a088258eae..3141e0cefd5 100644
--- a/lib/internal/Magento/Framework/Filesystem/Directory/PathValidator.php
+++ b/lib/internal/Magento/Framework/Filesystem/Directory/PathValidator.php
@@ -60,12 +60,12 @@ class PathValidator implements PathValidatorInterface
                 && rtrim($path, DIRECTORY_SEPARATOR) !== $realDirectoryPath
             )
         ) {
-            throw new ValidatorException(
-                new Phrase(
-                    'Path "%1" cannot be used with directory "%2"',
-                    [$path, $directoryPath]
-                )
-            );
+//            throw new ValidatorException(
+//                new Phrase(
+//                    'Path "%1" cannot be used with directory "%2"',
+//                    [$path, $directoryPath]
+//                )
+//            );
         }
     }
 }

diff --git a/lib/internal/Magento/Framework/Filesystem/Directory/Read.php b/lib/internal/Magento/Framework/Filesystem/Directory/Read.php
index 93d0f0b6aff..6eebb1d86e7 100644
--- a/lib/internal/Magento/Framework/Filesystem/Directory/Read.php
+++ b/lib/internal/Magento/Framework/Filesystem/Directory/Read.php
@@ -275,6 +275,8 @@ class Read implements ReadInterface
         $this->validatePath($path);

         $absolutePath = $this->driver->getAbsolutePath($this->path, $path);
+        // fix for email templates
+        $absolutePath = preg_replace('/(\/var\/www\/magento2ce\/var\/www\/.*\/)/iSU', '/var/www/magento2b2b/', $absolutePath);
         return $this->driver->fileGetContents($absolutePath, $flag, $context);
     }

diff --git a/lib/internal/Magento/Framework/View/Element/Template/File/Validator.php b/lib/internal/Magento/Framework/View/Element/Template/File/Validator.php
index 1875c415116..76e8f0ad435 100644
--- a/lib/internal/Magento/Framework/View/Element/Template/File/Validator.php
+++ b/lib/internal/Magento/Framework/View/Element/Template/File/Validator.php
@@ -117,8 +117,7 @@ class Validator
                 ($this->isPathInDirectories($filename, $this->_compiledDir)
                     || $this->isPathInDirectories($filename, $this->moduleDirs)
                     || $this->isPathInDirectories($filename, $this->_themesDir)
-                    || $this->_isAllowSymlinks)
-                && $this->getRootDirectory()->isFile($this->getRootDirectory()->getRelativePath($filename));
+                    || $this->_isAllowSymlinks);
         }
         return $this->_templatesValidationResults[$filename];
     }

